src/app/redux/hooks.ts

```diff
@@ -0,0 +1,7 @@
+"use client";
+
+import {  TypedUseSelectorHook,useDispatch, useSelector } from 'react-redux';
+import type { RootState, AppDispatch } from './store';
+
+export const useAppDispatch = () => useDispatch<AppDispatch>();
```

Comment:
The useAppDispatch as you've defined it may not work. You need to set its type to RootState to ensure it works as intended